### ISSUE TITLE
Enable Wayland support in the beta branch

### ISSUE THREAD

**Issue by Bryophyllum** [flathub/org.telegram.desktop#101]
With the recent changes made to `QGnomePlatform`, reverting https://github.com/flathub/org.telegram.desktop/commit/be6087d536c42048f9e1ad45979b8323dc7023a1, or other code equivalent that forces the `x11` backend to be used, to test the QT Wayland backend on soon-to-be-officially-released Fedora 31 should be prioritized.

**Comment by Bryophyllum** [comment]
I wrongly assumed that the theming would be correct by now:
![telegram_qt_wayland](https://user-images.githubusercontent.com/27376783/66723489-0dd2f400-ee1a-11e9-8930-091bdfbc9f6c.jpg)

Is there anything for Flatpak to adopt before having proper window decorations on QT Wayland under GNOME?

**Comment by grulja** [comment]
You need to install org.kde.WaylandDecoration.QGnomePlatform-decoration.

**Comment by grulja** [comment]
I think it is a good idea to enable wayland on the beta branch. I did that. You can expect a new update soon.

**Comment by Bryophyllum** [comment]
It was already installed on my system; however, how I forced Qt to use the Wayland backend might have something to do with the wrong decorations: `XDG_CURRENT_DESKTOP=KDE flatpak run org.telegram.desktop`

**Comment by Bryophyllum** [comment]
The update was just pushed to flathub-beta, and the wrong window decoration is still used.

I think it needs to be re-built against Qt 5.13:
```
$ flatpak info org.kde.WaylandDecoration.QGnomePlatform-decoration

QGnomePlatform-decoration - Qt Platform Theme aimed to accommodate GNOME
settings

          ID: org.kde.WaylandDecoration.QGnomePlatform-decoration
         Ref: runtime/org.kde.WaylandDecoration.QGnomePlatform-decoration/x86_64/5.12
        Arch: x86_64
      Branch: 5.12
      Origin: flathub
  Collection: org.flathub.Stable
Installation: system
   Installed: 156.7 kB

      Commit: e05804ace14fd1f2278c8eb83ef2564decd962337fd0abbce829fd2cbf3f1c71
     Subject: Fix appstream id (527cbfb9)
        Date: 2019-08-16 19:14:59 +0000 
```

**Comment by grulja** [comment]
Ups, I updated qgnomeplatform and forgot to update decorations. Pushed. Thanks.

**Comment by Bryophyllum** [comment]
It still uses Qt's window decoration even after having `org.kde.WaylandDecoration.QGnomePlatform-decoration` updated to 5.13

**Comment by grulja** [comment]
You mentioned above that you use `XDG_CURRENT_DESKTOP=KDE flatpak run org.telegram.desktop`. With that it won't load QGnomePlatform, you will need to use `GNOME` instead.

**Comment by Bryophyllum** [comment]
I used that to launch Telegram before https://github.com/flathub/org.telegram.desktop/commit/4aec7e507a64187f4783de14fc3fcaa095a35004 was pushed to the `beta` branch.
Currently I launch it using an xdg-desktop entry pinned to my dock.

**Comment by Bryophyllum** [comment]
I am able to force the flatpak to use `QGnomePlatform` with `$ QT_QPA_PLATFORMTHEME='gnome' flatpak run org.telegram.desktop`. It appears that Qt does not pick the platform theme up automatically, but otherwise it works (_though, if I am allowed to chime in my personal opinion, the decoration is far from perfect, but it is clearly better than the default decorations, especially when its quirks, such as a upscaled mouse cursor when hovering over Telegram's window are considered_).

**Comment by Bryophyllum** [comment]
@grulja Is this being worked on? Should I open a new issue? Without overriding the platform theme, Qt's default one is still used by Telegram instead, even though my `$XDG_CURRENT_DESKTOP` yields `GNOME`.

**Comment by grulja** [comment]
Can you add output from_
`QT_DEBUG_PLUGINS=1 flatpak run org.telegram.desktop`

**Comment by Bryophyllum** [comment]
https://gist.github.com/Bryophyllum/9d10d4dc947a3c2441424c8d07d8a4d9

**Comment by grulja** [comment]
Let's make it clear what you are doing.
1) Are you on GNOME?
2) What do you do with Telegram to open the dialog?

From the log I see that XdgDesktopPortal platformtheme is loaded, which is correct, because it's loaded based on app running in Flatpak. With that, any "Open/Save file" attempt should open a dialog outside the sandbox, which means using portal dialog from xdg-desktop-portal-gtk (in case you are in GNOME).

**Comment by Bryophyllum** [comment]
1. I'm on Fedora Silverblue 31 running GNOME 3.34.1 on Wayland
2. I right-click on an image, and choose `Save Image As...` 

There looks like to be a confusion, let me clear it up. The GTK dialog is properly themed, but the window decoration, as exampled in the comment above https://github.com/flathub/org.telegram.desktop/issues/101#issuecomment-541466422, does not match the Adwaita theme.

You are probably thinking that I experience the issue https://github.com/flathub/org.telegram.desktop/issues/102, which, as per the comment https://github.com/flathub/org.telegram.desktop/issues/102#issuecomment-542042725, stemmed from an explicit platform theme override.

To summarize: I am on GNOME 3.34.1 running on Wayland, I do not override the platform theme, the GTK file picker dialog is fine, but the window decoration does not match the look and feel of Adwaita:
![telegram_wrong_decoration](https://user-images.githubusercontent.com/27376783/67096710-fba9da80-f1b8-11e9-8c57-5757e4ac3e65.png)

**Comment by grulja** [comment]
I understand now your problem, but I'm not able to reproduce it. I will get "GNOME"decorations automatically when opening telegram in GNOME wayland session. The only problem I found is that with Qt 5.13, when we removed dconf access, we need additional portal support for the decorations to pick correct button layout and font for the window title, but that's a bug in QGnomePlatform.

**Comment by Bryophyllum** [comment]
I am able to reproduce it on a virtual machine running Fedora Silverblue 31 using GNOME Boxes.

**Steps to reproduce:**

1. Install GNOME Boxes; it is available on `flathub`
2. Click on the plus icon in the left upper corner, and choose "Create a Virtual Machine"
3. Select "Fedora Silverblue 30 x86_64" from a list of sources
4. Perform an Express Install, and leave the default properties of the VM unaltered
5. After a successful installation, rebase the OS to Fedora Silverblue 31 testing:

- `$ rpm-ostree rebase fedora:fedora/31/x86_64/testing/silverblue`
- `$ reboot`

6. Then add appropiate Flatpak remotes, and install Telegram:

- `$ flatpak remote-add flathub-beta https://flathub.org/beta-repo/flathub-beta.flatpakrepo`
- `$ flatpak remote-add flathub https://flathub.org/repo/flathub.flatpakrepo`
- `$ flatpak install org.telegram.desktop//beta`

7. Run Telegram

- `$ flatpak run org.telegram.desktop`

**Comment by grulja** [comment]
Are you sure you have both org.kde.PlatformTheme.QGnomePlatform and org.kde.WaylandDecoration.QGnomePlatform-decoration installed? I'm testing this on Fedora 31.

**Comment by Bryophyllum** [comment]
It is installed on the virtual machine:

![Screenshot from 2019-10-19 16-17-35](https://user-images.githubusercontent.com/27376783/67146509-08e3c980-f28c-11e9-86a8-67573c5ec694.png)

..as well as on my host:
```
$ flatpak list
(...)
Adwaita dark GTK theme                 org.gtk.Gtk3theme.Adwaita-dark                                 3.22   flathub         system
Adwaita theme                          org.kde.KStyle.Adwaita                                         5.13   flathub         system
KDE Application Platform               org.kde.Platform                                               5.12   flathub         system
KDE Application Platform               org.kde.Platform                                               5.13   flathub         system
QGnomePlatform                         org.kde.PlatformTheme.QGnomePlatform                           5.12   flathub         system
QGnomePlatform                         org.kde.PlatformTheme.QGnomePlatform                           5.13   flathub         system
QGnomePlatform-decoration              org.kde.WaylandDecoration.QGnomePlatform-decoration            5.12   flathub         system
QGnomePlatform-decoration              org.kde.WaylandDecoration.QGnomePlatform-decoration            5.13   flathub         system
(...)

**Comment by grulja** [comment]
I still cannot reproduce it ony my Fedora 31, even two of my collegues tried it on Fedora Workstation and Fedora Silverblue and none of them had this problem.

**Comment by Bryophyllum** [comment]
Hello @grulja. This problem is still reproducible on a fresh installation of Fedora 31 Silverblue.

**Steps to reproduce** are same as in https://github.com/flathub/org.telegram.desktop/issues/101#issuecomment-544134161, but Fedora 31 Silveblue is installed from an ISO instead and thus re-basing the OS is unnecessary:

1. _Install GNOME Boxes (it is available on `flathub`), and open it_
2. _Click on the plus icon in the left upper corner, and choose "Create a Virtual Machine"_
3. **Select "Fedora Silverblue 31" from a list of sources**
4. _Perform an Express Install, and leave the default properties of the VM unaltered_
5. _Then add appropiate Flatpak remotes, and install Telegram:_

- _`$ flatpak remote-add flathub-beta https://flathub.org/beta-repo/flathub-beta.flatpakrepo`_
- _`$ flatpak remote-add flathub https://flathub.org/repo/flathub.flatpakrepo`_
- _`$ flatpak install org.telegram.desktop//beta`_

7. _Run Telegram:_

- _`$ flatpak run org.telegram.desktop`_


![Screenshot from 2019-11-30 18-02-31](https://user-images.githubusercontent.com/27376783/69903683-26cd3f80-139d-11ea-8321-410c396d230f.png)

**Comment by Bryophyllum** [comment]
@grulja A recent Mutter/GNOME/Telegram update has solved this window decoration issue for me. 

Looking forward to seeing the Wayland support pushed to the stable branch. Thank you for your hard work!
